src/store/auth.ts

```diff
@@ -1,19 +1,20 @@
 import { defineStore } from 'pinia';
-import { login, register, setAuthHeader, removeAuthHeader } from '@/services/api/auth.js';
+import { login, register, setAuthHeader, removeAuthHeader } from '@/services/api/auth';
+import { LoginCredentials, RegisterCredentials } from '@/types/auth';
 
 const TOKEN_STORAGE_KEY = 'auth-token';
 
 export const useAuthStore = defineStore('auth', {
   state() {
     return {
-      token: null,
+      token: null as string,
```

Comment:
Null isn't a string - you could set it by default as an empty string (it's a falsy value) or perhaps mark this as `string | null` 😄 
```suggestion
      token: '',
```

Reply:
About to be `''` 😄